tests/pytorch/test_stream.py

```diff
@@ -126,14 +127,18 @@ def perform_computing():
     with torch.cuda.stream(stream):
         # since we have called record stream for g2, g3 won't reuse its memory
         g3 = rand_graph(10, 20, device=F.ctx())
+        g3.create_formats_()
     torch.cuda.current_stream().synchronize()
     assert torch.equal(result, results2)
 
 @unittest.skipIf(F._default_context_str == 'cpu', reason="stream only runs on GPU.")
+# this test must follow test_record_stream_graph_positive to ensure
+# enough memory blocks have been reserved.
```

Comment:
Could you please elaborate more on the comment here? 
What if you release all reserved memory by `empty_cache()` before negative test. Would it fail?

Reply:
Running `test_record_stream_graph_negative()` solely will fail. I don't know why...

Reply:
I profiled it and it seems from the sync attributed to `cusparsecreate()` call (invoking cudaFree() underneath) when we first calling spMM function. https://github.com/dmlc/dgl/blob/d78a3a4baf611b90871a849f58647160c7cd9ab4/src/array/cuda/spmm.cuh#L231 

And because of that sync, if we just call negative test, computation cannot really overlap with data cpy and it will always fail.

Reply:
I have updated both the positive and negative tests to initialize the cusparse handle before computing.